diff --git a/node_modules/expo/android/build.gradle b/node_modules/expo/android/build.gradle
index 5645b26..fbe894a 100644
--- a/node_modules/expo/android/build.gradle
+++ b/node_modules/expo/android/build.gradle
@@ -14,9 +14,10 @@ static def versionToNumber(major, minor, patch) {
 }
 
 def getRNVersion() {
+  def nodeExecutable = rootProject.findProperty("NODE_BINARY") ?: "node"
   def nodeModulesVersion = providers.exec {
     workingDir(projectDir)
-    commandLine("node", "-e", "console.log(require('react-native/package.json').version);")
+    commandLine(nodeExecutable, "-e", "console.log(require('react-native/package.json').version);")
   }.standardOutput.asText.get().trim()
 
   def version = safeExtGet("reactNativeVersion", nodeModulesVersion)
