diff --git a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
index bcf9b00..dc7d6e7 100644
--- a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
+++ b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
@@ -80,9 +80,12 @@ ext.useExpoPublishing = {
 
   project.afterEvaluate {
     publishing {
-      publications {
-        release(MavenPublication) {
-          from components.release
+      def releaseComponent = components.findByName('release')
+      if (releaseComponent != null) {
+        publications {
+          release(MavenPublication) {
+            from releaseComponent
+          }
         }
       }
       repositories {
diff --git a/node_modules/expo-modules-core/android/build.gradle b/node_modules/expo-modules-core/android/build.gradle
index 23eb20f..db594fc 100644
--- a/node_modules/expo-modules-core/android/build.gradle
+++ b/node_modules/expo-modules-core/android/build.gradle
@@ -24,7 +24,7 @@ def REACT_NATIVE_DIR = REACT_NATIVE_BUILD_FROM_SOURCE
   ? findProject(":packages:react-native:ReactAndroid").getProjectDir().parent
   : file(providers.exec {
       workingDir(rootDir)
-      commandLine("node", "--print", "require.resolve('react-native/package.json')")
+      commandLine(rootProject.findProperty("NODE_BINARY") ?: "node", "--print", "require.resolve('react-native/package.json')")
     }.standardOutput.asText.get().trim()).parent
 
 def reactProperties = new Properties()
