diff --git a/node_modules/expo-constants/scripts/get-app-config-android.gradle b/node_modules/expo-constants/scripts/get-app-config-android.gradle
index 915875a..8101808 100644
--- a/node_modules/expo-constants/scripts/get-app-config-android.gradle
+++ b/node_modules/expo-constants/scripts/get-app-config-android.gradle
@@ -2,14 +2,15 @@
 
 import org.apache.tools.ant.taskdefs.condition.Os
 
+def nodeExecutable = rootProject.findProperty("NODE_BINARY") ?: "node"
 
 def expoConstantsDir = project.providers.exec {
   workingDir(projectDir)
-  commandLine("node", "-e", "console.log(require('path').dirname(require.resolve('expo-constants/package.json')));")
+  commandLine(nodeExecutable, "-e", "console.log(require('path').dirname(require.resolve('expo-constants/package.json')));")
 }.standardOutput.asText.get().trim()
 
 def config = project.hasProperty("react") ? project.react : [];
-def nodeExecutableAndArgs = config.nodeExecutableAndArgs ?: ["node"]
+def nodeExecutableAndArgs = config.nodeExecutableAndArgs ?: [nodeExecutable]
 
 afterEvaluate {
   def projectRoot = file("${rootProject.projectDir}")
